extensions/Cake.Issues.DocFx.yml

```diff
@@ -3,7 +3,8 @@ Name: Cake.Issues.DocFx
 NuGet: Cake.Issues.DocFx
 Assemblies:
 - "/**/Cake.Issues.DocFx.dll"
-Repository: https://github.com/cake-contrib/Cake.Issues.DocFx/
+Repository: https://github.com/cake-contrib/Cake.Issues.DocFx.git
+ProjectUrl: https://github.com/cake-contrib/Cake.Issues.DocFx/
```

Comment:
@Jericho ProjctUrl in this case should be https://cakeissues.net/ (see https://www.nuget.org/packages/Cake.Issues.DocFx/0.9.0, where both project URL and repo are available in metadata)

Reply:
I see. I just investigated and found the source of this problem to be the logic that checks if the repository was trasnfered to cake-contrib and if so, replace the projectUrl and the repositoryUrl with the URLs in the cake-contrib organization. This was done because some addin authors still have the URL pointing to the repo under their personal GitHub account and we wanted to ensure the proper URL to be used.

I could remove this logic which would fix the problem for Cake.Issues.DocFx but it would cause the projectUrl for a number of other addins to point to the wrong location.

Which means that I need to spend a little more time thinking about we can possibly fix this.

Reply:
If it is either/or, IMHO we should use the project URL as is and disable to auto-correction.

Reply:
If I remove the auto-correct logic, it solves the problem for Cake.Issues.DocFx (and probably a few other addins as well) but it will create another problem: the nuspec for some addins contains the projectURL for the author's personal account (which was correct before transferring the repo to cake-contrib) but this URL is no longer valid because the repo has been transferred to the cake-contrib organization. In these cases, the URL in the resulting YAML would be incorrect.

Reply:
I'm aware of the consequences. We should never change correct data to something which is not correct, just to try to fix some wrong data. If we need to decide between these two I prefer the project URls pointing to the wrong location (which it also does on nuget.org) over having them fixed and using wrong project URLs for the addins where the author has put in a correct project URL.

Reply:
Should bring @gep13 in this discussion because correcting the projectUrl in the various reports that AddinDiscoverer generates was something he [requested](https://github.com/cake-contrib/Cake.AddinDiscoverer/issues/14).

Also, [other people](https://github.com/cake-contrib/Cake.AddinDiscoverer/issues/102) rely of this URL being valid.

Like I said, I need to spend a little more time to figure out how we can possibly fix this without creating other problems.

Reply:
If there's a possibility to have both, this is the way to go of course. My comment was in case if there is no possibility to have both, we should decide to take the correct values, instead of fixing wrong values and making correct values wrong.

Or we don't take these values from Addin Discoverer if the logic for the auditing needs to be different for some people there and instead read them as part of the website code (we also have the metadata of the NuGet package available there).

Reply:
> If there's a possibility to have both

That's definitely my goal.

Reply:
By the way, the offending logic is [here](https://github.com/cake-contrib/Cake.AddinDiscoverer/blob/develop/Source/Cake.AddinDiscoverer/Steps/ValidateUrlStep.cs#L29-L36).

How about we overwrite the ProjectUrl only if it's a GitHub or Bitbucket URL and we preserve custom URLs such as `https://cakeissues.net/`?

Reply:
This could work. github.io URL should also be preserved, since some addins have documentation there

Reply:
Yes, I already consider `.io` as well as `.com`:

```csharp
public static bool IsGithubUrl(this Uri uri)
{
    if (uri == null) return false;

    return uri.Host.Contains("github.com", StringComparison.OrdinalIgnoreCase) ||
           uri.Host.Contains("github.io", StringComparison.OrdinalIgnoreCase);
}

public static bool IsBitbucketUrl(this Uri uri)
{
    if (uri == null) return false;

    return uri.Host.Contains("bitbucket.org", StringComparison.OrdinalIgnoreCase);
}
```

Reply:
oh wait I just re-read your comment and you are actually suggesting to PRESERVE github.io URLs. Didn't think about that. I was actually going to do exactly the opposite and overwrite github.io URLs.

Reply:
If I understand it correctly this means that github.io link could also be changed which should not be the case. A github.io URL should be treated like any custom URL.

Reply:
I agree with you to preserve github.io URLs.

Reply:
Good news: I implemented the logic to preserve custom URLs and github.io URLs means means that `https://cakeissues.net/` is preserved. 

Other good news: this logic also preserves the documentation URL of some addins that was previously overwritten such as `https://cake-contrib.github.io/Cake.AzureDevOps/` for example.

Bad news: the projetUrl for some addins points to `github.io` (and therefore we preserve it) but it turns out to be invalid such as `https://cake-contrib.github.io/Cake.AsciiDoctorJ/` for example.

So I made some progress but I have yet one more problem to resolve.

Reply:
In these cases where the metadata is wrong we should try to fix it in the addins, especially for the ones which are part of cake-contrib.

Reply:
Closing this discussion and publishing new version of the discoverer